packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -2,20 +2,921 @@
+    final BuildContext? context = anchorKey.currentContext;
+    if (context != null) {
+      final RenderBox searchBarBox = context.findRenderObject()! as RenderBox;
+      final Size boxSize = searchBarBox.size;
+      final Offset boxLocation = searchBarBox.localToGlobal(Offset.zero);
+      return boxLocation & boxSize;
+    }
+    return null;
+  }
+
+  @override
+  TickerFuture didPush() {
+    assert(anchorKey.currentContext != null);
+    updateViewConfig(anchorKey.currentContext!);
+    updateTweens(anchorKey.currentContext!);
+    toggleVisibility?.call();
+    return super.didPush();
+  }
+
+  @override
+  bool didPop(_SearchViewRoute? result) {
+    assert(anchorKey.currentContext != null);
+    updateTweens(anchorKey.currentContext!);
+    toggleVisibility?.call();
+    return super.didPop(result);
+  }
+
+  void updateViewConfig(BuildContext context) {
+    viewDefaults = _SearchViewDefaultsM3(context, isFullScreen: showFullScreenView);
+    viewTheme = SearchViewTheme.of(context);
+    dividerTheme = DividerTheme.of(context);
+  }
+
+  void updateTweens(BuildContext context) {
+    final Size screenSize = MediaQuery.of(context).size;
+    final Rect anchorRect = getRect() ?? Rect.zero;
+
+    // Check if the search view goes off the screen.
+    final BoxConstraints effectiveConstraints = viewConstraints ?? viewTheme.constraints ?? viewDefaults.constraints!;
+    final double verticalDistanceToEdge = screenSize.height - anchorRect.top;
+    final double endHeight = math.max(effectiveConstraints.minHeight, math.min(screenSize.height * 2 / 3, verticalDistanceToEdge));
+    _rectTween.begin = anchorRect;
+
+    switch (textDirection ?? TextDirection.ltr) {
+      case TextDirection.ltr:
+        final double viewEdgeToScreenEdge = screenSize.width - anchorRect.left;
+        final double endWidth = math.max(effectiveConstraints.minWidth, math.min(anchorRect.width, viewEdgeToScreenEdge));
+        final Size endSize = Size(endWidth, endHeight);
+        _rectTween.end = showFullScreenView ? Offset.zero & screenSize : (anchorRect.topLeft & endSize);
+        break;
+      case TextDirection.rtl:
+        final double viewEdgeToScreenEdge = anchorRect.right;
+        final double endWidth = math.max(effectiveConstraints.minWidth, math.min(anchorRect.width, viewEdgeToScreenEdge));
+        final Offset topLeft = Offset(math.max(anchorRect.right - endWidth, 0.0), anchorRect.top);
+        final Size endSize = Size(endWidth, endHeight);
+        _rectTween.end = showFullScreenView ? Offset.zero & screenSize : (topLeft & endSize);
+    }
+  }
+
+  @override
+  Widget buildPage(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+
+    return Directionality(
+      textDirection: textDirection ?? TextDirection.ltr,
+      child: AnimatedBuilder(
+        animation: animation,
+        builder: (BuildContext context, Widget? child) {
+          final Animation<double> curvedAnimation = CurvedAnimation(
```

Comment:
https://api.flutter.dev/flutter/animation/CurvedAnimation/reverseCurve.html
> If you use a non-null reverseCurve you might want to hold this object in a State object rather than recreating it each time your widget builds in order to take advantage of the state in this object that avoids visual discontinuities.

If bringing the page up and down (before the animation completes) looks janky, making the CurvedAnimation part of this Route's state might be a good idea. Could be handled in a separate PR.